runtime/stringbuf.c

```diff
@@ -281,6 +281,10 @@ rsRetVal rsCStrAppendStrWithLen(cstr_t *pThis, const uchar* psz, size_t iStrLen)
 	memcpy(pThis->pBuf + pThis->iStrLen, psz, iStrLen);
 	pThis->iStrLen += iStrLen;
 
+	/* reset any cached sz string */
+	free(pThis->pszBuf);
```

Comment:
I cannot look at this in-depth at the moment, but some piece of information: after cstrFinalize (or similar name) has been called, the buffer should be read-only. It's not being enforced, but this may be the root cause of the issue. I'll look in more depth when I am finished with the higher priority things.